homeassistant/components/esphome/button.py

```diff
@@ -42,10 +42,16 @@ def _on_static_info_update(self, static_info: EntityInfo) -> None:
 
     @callback
     def _on_device_update(self) -> None:
-        """Update the entity state when device info has changed."""
-        # This override the EsphomeEntity method as the button entity
-        # never gets a state update.
-        self._on_state_update()
+        """Call when device updates or entry data changes.
+
+        Overwrites the default behavior to not write state, but
```

Comment:
The comment mentions that we're not writing state, and then calls the write state function.

Reply:
Oh never mind, I read it again and get it now.

Reply:
I rewrote the whole comment so it makes more sense.  I did a couple refactor passes until I got it right and the comment was an amalgamation of iterations